openwisp_utils/base.py

```diff
@@ -3,23 +3,38 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 from model_utils.fields import AutoCreatedField, AutoLastModifiedField
+from openwisp_utils.utils import get_random_key
+from openwisp_utils.validators import key_validator
 
 
-class TimeStampedEditableModel(models.Model):
+class UUIDModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class TimeStampedEditableModel(UUIDModel):
     """
     An abstract base class model that provides self-updating
     ``created`` and ``modified`` fields.
     """
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     created = AutoCreatedField(_('created'), editable=True)
     modified = AutoLastModifiedField(_('modified'), editable=True)
 
     class Meta:
         abstract = True
 
 
-class UUIDModel(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-
-    class Meta:
-        abstract = True
+class KeyField(models.CharField):
```

Comment:
@nemesisdesign I really don't want having to write `CharField` with all those parameters by hand every time. How about this custom field which is just a `CharField` with some default params?